### PR TITLE
feat(ai-eval): Spec 69 Phase 4 — anomaly/pattern/watcher scenario adapters + 64 fixtures

### LINKED ISSUE #351: Spec 69 Phase 4 — Scenario expansion (anomaly + pattern + watcher fixtures)
Placeholder. Detailed scope intentionally deferred.

**Mandate (per spec 69 §11 + §3.1)**: Author fixtures + scenario adapters for the three remaining AI scenarios in cap-ai-provider:
- anomaly-detector (≥20 fixtures, ≥4 tags)
- pattern-detector (≥20 fixtures, ≥4 tags)
- watcher-engine AI-driven triggers (≥15 fixtures, ≥3 tags)

**Blocked by**: #348 (framework + scenario adapter pattern proven on intent first), Phase 2 issue (runner choice fixed)

**Out of scope**: red-team / security fixtures — spec 27 owns those (per spec 69 §10.4).

### PR DESCRIPTION
**Pull request by laofahai**
## Summary

- Add three deterministic `ScenarioAdapter` implementations in `cap-ai-provider/eval-runner/`: `anomaly-detector-scenario`, `pattern-detector-scenario`, `watcher-engine-scenario`
- Ship 64 fixture JSON files (anomaly×25, pattern×22, watcher×17) across categorised sub-directories
- Add 38 unit tests (all passing) covering golden paths, edge cases, disabled watchers, tenant filtering, and replay determinism
- Extend `@linchkit/devtools` types with a proper `WatcherTriggerInput` discriminated union and three new matcher catalogs (anomaly, pattern, watcher)

## Implementation notes

**AnomalyDetector adapter** — converts `AnomalyUsageEventInput[]` to `AnomalyUsageEvent`, feeds them into a fresh `AnomalyDetector` instance per run. Forwards `windowSizeMs` and `tenantId` from fixture context. Returns serialisable anomaly items.

**PatternDetector adapter** — hydrates an `ExecutionLogger` with entries, mapping `entry.timestamp → startedAt: Date` (the `ExecutionLogEntry` field name). Strips non-deterministic IDs (based on `Date.now()`) when comparing replay output.

**WatcherEngine adapter** — converts `WatcherDefInput` to `WatcherDefinition` via a discriminated-union `buildTrigger()` helper, creates a fresh registry + engine per run, calls `evaluateAfterMutation(entityName, record, oldRecord)`. Marks `schedule` / `staleness` triggers as `trigger_type_not_reactive` since they are not post-mutation reactive.

**DeclarativeCondition field paths** — discovered that `resolveField()` starts from `ConditionContext` (not `ctx.target`), so filter conditions in fixture files must use the `"target.fieldName"` prefix (e.g. `{ field: "target.status", operator: "eq", value: "active" }`).

**AnomalyDetector window** — default `windowSizeMs` is 300,000 ms (5 min), not 1 hr; all unit tests pass an explicit `windowSizeMs` and space events accordingly.

## Spec alignment

Implements Spec 69 Phase 4 (scenario adapter registry, deterministic adapters, fixture-driven eval). The `ScenarioAdapter` interface contract (`runLive` + `replayFromBaseline`) is satisfied by all three adapters.

## Quality gates

```
bun run check  → 0 errors (schema-version info only)
bun test addons/ai-provider/cap-ai-provider/__tests__/  → 233 pass, 0 fail
```

(Pre-existing missing-dependency failures in other packages are unrelated to this PR.)

## Retro

- `windowSizeMs` default discovery required direct debugging — worth documenting in the AnomalyDetector docstring
- `resolveField`'s `"target."` prefix requirement for `DeclarativeCondition` is a subtle contract; a unit test in the core engine would prevent future confusion
- `PatternDetector` ID non-determinism (Date.now()) makes replay comparison slightly awkward — a future refactor could accept an ID factory

## Self-review checklist

- [x] No LLM calls in any adapter (fully deterministic)
- [x] No hardcoded secrets or `any` types
- [x] All new public symbols exported from `@linchkit/devtools` barrel
- [x] Fixture JSON files are valid (biome formatted)
- [x] `replayFromBaseline` returns structurally equivalent output to `runLive`
- [x] Disabled watchers and wrong-entity watchers correctly return empty/no-fire results

Closes #351

https://claude.ai/code/session_01FemhjbhKZKxraR3ZrNVGep

---
_Generated by [Claude Code](https://claude.ai/code/session_01FemhjbhKZKxraR3ZrNVGep)_

## Summary by CodeRabbit

# Release Notes

* **Tests**
  * Added comprehensive test suites for anomaly detection, pattern detection, and watcher engine scenario evaluations
  * Added numerous test fixtures covering various detection scenarios and edge cases for each feature

* **New Features**
  * Introduced scenario adapters and matcher functions for anomaly detection, pattern detection, and watcher engine evaluations
  * Added type definitions and public exports for the evaluation framework

[Review Change Stack](https://app.coderabbit.ai/change-stack/laofahai/linchkit/pull/383?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: packages/devtools/src/ai-eval/matchers/watcher.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+/** Assert that a specific watcher fired. */
+const watcher_fired: MatcherFn<WatcherEvalOutput> = (output, args) => {
+  if (typeof args.name !== "string") return fail("watcher_fired", "arg 'name' must be a string");
+  const entry = output.find((w: WatcherEvalOutputItem) => w.watcherName === args.name);
+  if (!entry) return fail("watcher_fired", `watcher "${args.name}" not found in results`);
+  if (entry.fired) return pass("watcher_fired", args.name);
+  return fail(
+    "watcher_fired",
+    `watcher "${args.name}" did not fire: ${entry.reason ?? "unknown reason"}`,
+  );
```
[the remark this answers is not in the document]
Comment (laofahai) [reply packages/devtools/src/ai-eval/matchers/watcher.ts:38]: Fixed in commit 3aeca1c — `watcher_fired` now reads `args.watcherName` to match all fixture files.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/devtools/src/ai-eval/matchers/watcher.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+};
+
+/** Assert that a specific watcher did NOT fire. */
+const watcher_not_fired: MatcherFn<WatcherEvalOutput> = (output, args) => {
+  if (typeof args.name !== "string")
+    return fail("watcher_not_fired", "arg 'name' must be a string");
+  const entry = output.find((w: WatcherEvalOutputItem) => w.watcherName === args.name);
+  if (!entry) return pass("watcher_not_fired");
+  if (!entry.fired) return pass("watcher_not_fired");
+  return fail("watcher_not_fired", `watcher "${args.name}" fired but was expected not to`);
```
[the remark this answers is not in the document]
Comment (laofahai) [reply packages/devtools/src/ai-eval/matchers/watcher.ts]: Fixed in commit 3aeca1c — `watcher_not_fired` now reads `args.watcherName` to match all fixture files.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: addons/ai-provider/cap-ai-provider/eval-runner/pattern-detector-scenario.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    config.lookbackDays = lookbackDays;
+  }
+
+  // Override the internal "now" via config — PatternDetector computes the
+  // lookback window as `new Date()` minus `lookbackDays`. To make fixtures
+  // deterministic regardless of when they run, we set lookbackDays to a very
+  // large value (3650 = 10 years) so ALL entries are always in the window.
+  const detector = new PatternDetector({
+    ...config,
+    lookbackDays: config.lookbackDays ?? 3650,
```
[the remark this answers is not in the document]
Comment (laofahai) [reply addons/ai-provider/cap-ai-provider/eval-runner/pattern-detector-scenario.ts:48]: Fixed in commit 3aeca1c — removed the `if (context?.now)` branch that capped lookbackDays at 30, and stopped mutating `input.config`. Now always defaults to 3650 days via `config.lookbackDays ?? 3650`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: addons/ai-provider/cap-ai-provider/eval-runner/pattern-detector-scenario.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+export function createPatternDetectorScenario(): PatternDetectorScenarioAdapter {
+  return {
+    async runLive(fx) {
+      return runPatternDetector(fx.input, fx.context);
+    },
+    replayFromBaseline(fx, _baseline) {
+      // Deterministic: re-run the detector. Baseline not needed.
+      return runPatternDetector(fx.input, fx.context) as unknown as PatternEvalOutput;
+    },
```
[the remark this answers is not in the document]
Comment (laofahai) [reply addons/ai-provider/cap-ai-provider/eval-runner/pattern-detector-scenario.ts]: Fixed in commit 3aeca1c — removed the `as unknown as PatternEvalOutput` cast; `replayFromBaseline` now returns the Promise directly.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/devtools/src/ai-eval/types.ts [hunk 5]
```diff
@@ -246,6 +246,184 @@ export interface RunReport<TOutput = unknown> {
...
+
+/** Output shape returned by the pattern-detector scenario adapter. */
+export type PatternEvalOutput = PatternEvalOutputItem[];
+
+// ── Watcher Engine scenario types (Spec 69 Phase 4) ─────────────────────
+
+/** Threshold trigger input (JSON-serialisable). */
+export interface WatcherThresholdTriggerInput {
+  type: "threshold";
+  field?: string;
```
[the remark this answers is not in the document]
Comment (laofahai) [reply packages/devtools/src/ai-eval/types.ts]: Fixed in commit 3aeca1c — `WatcherThresholdTriggerInput.field` is now required, consistent with `WatcherStalenessTriggerInput`.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by laofahai** [comment]
All 5 issues flagged by Gemini have been addressed in commit `3aeca1c`:

1. `watcher_fired` / `watcher_not_fired` — now use `args.watcherName` (was `args.name`) to match all fixture files
2. Pattern-detector lookback — removed the `if (context?.now)` branch that capped lookbackDays at 30; always defaults to 3650-day window now
3. `replayFromBaseline` cast — removed the `as unknown as PatternEvalOutput` type lie; returns the Promise directly
4. `WatcherThresholdTriggerInput.field` — now required (was optional)

The **Review Threads** gate is still failing because the Gemini threads haven't been resolved. I'm unable to resolve them programmatically (the MCP tool requires GraphQL thread node IDs not returned by the API). Could you resolve the 5 Gemini review threads manually? Once they're all resolved, you can post `/check-reviews` to re-run the gate, or it will re-evaluate on the next push.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/devtools/src/ai-eval/matchers/watcher.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (laofahai) [reply packages/devtools/src/ai-eval/matchers/watcher.ts:38]: Addressed — `watcher_fired` and `watcher_not_fired` both read `args.watcherName` (matchers/watcher.ts:30,43), matching the fixture convention, and return a typed failure when it is missing. All 17 watcher fixtures + the ai-eval suite (84 tests) pass.

File: packages/devtools/src/ai-eval/matchers/watcher.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (laofahai) [reply packages/devtools/src/ai-eval/matchers/watcher.ts]: Addressed with the above — `watcher_not_fired` reads `args.watcherName` and now FAILS when the named watcher is absent from results (previously a vacuous pass that could hide fixture typos).

File: addons/ai-provider/cap-ai-provider/eval-runner/pattern-detector-scenario.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (laofahai) [reply addons/ai-provider/cap-ai-provider/eval-runner/pattern-detector-scenario.ts:48]: Addressed — the adapter no longer mutates `input.config`; it builds a fresh config object (`lookbackDays: config.lookbackDays ?? 3650`). `PatternDetector` has no injectable clock (it computes `new Date() - lookbackDays`), so `context.now` cannot deterministically anchor its window — the large default preserves deterministic replay and the unused `context` is marked `_context`.

File: addons/ai-provider/cap-ai-provider/eval-runner/pattern-detector-scenario.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (laofahai) [reply addons/ai-provider/cap-ai-provider/eval-runner/pattern-detector-scenario.ts]: Fixed — `replayFromBaseline`/`runLive` now `await runPatternDetector(...)` with no cast. `ScenarioAdapter.replayFromBaseline` was widened to `TOutput | Promise<TOutput>` and the runner awaits it, so async adapters hand matchers resolved output instead of an unresolved Promise.

File: packages/devtools/src/ai-eval/types.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (laofahai) [reply packages/devtools/src/ai-eval/types.ts]: Moot — `WatcherThresholdTriggerInput.field` is `field: string` (required, types.ts:350), matching all threshold fixtures and the trigger builder. There is no optionality left to tighten.
